website/src/components/McapRecordingDemo/McapRecordingDemo.tsx

```diff
@@ -0,0 +1,405 @@
+    return () => {
+      window.removeEventListener(
+        "deviceorientation",
+        handleDeviceOrientationEvent
+      );
+    };
+  }, [addPoseMessage, recording, recordOrientation]);
+
+  useEffect(() => {
+    const video = videoRef.current;
+    if (!recording || !recordVideo || !video) {
+      return;
+    }
+
+    setVideoStarted(false);
+    setVideoPermissionError(false);
+
+    const stopCapture = startVideoCapture({
+      video,
+      frameDurationSec: 1 / 30,
+      onStart: () => setVideoStarted(true),
+      onError: (err) => {
+        console.error(err);
+        setVideoPermissionError(true);
+      },
+      onFrame: (blob) => addCameraImage(blob),
+    });
+    return () => {
+      stopCapture();
+    };
+  }, [addCameraImage, recordVideo, recording]);
+
+  const onRecordClick = useCallback(
+    (event: React.MouseEvent) => {
+      event.preventDefault();
+      if (recording) {
+        setRecording(false);
+        return;
+      }
+      setRecording((oldValue) => !oldValue);
+
+      // Requesting orientation permission must be done as part of a user gesture
+      setOrientationPermissionError(false);
+      if (
+        recordOrientation &&
+        typeof DeviceOrientationEvent !== "undefined" &&
+        "requestPermission" in DeviceOrientationEvent &&
+        typeof DeviceOrientationEvent.requestPermission === "function"
+      ) {
+        void Promise.resolve(DeviceOrientationEvent.requestPermission())
+          .then((result) => {
+            if (result !== "granted") {
+              setOrientationPermissionError(true);
+            }
+          })
+          .catch(console.error);
+      }
+    },
+    [recordOrientation, recording]
+  );
+
+  const onDownloadClick = useCallback(
+    (event: React.MouseEvent) => {
+      event.preventDefault();
+      void (async () => {
+        const blob = await state.closeAndRestart();
+        const url = URL.createObjectURL(blob);
+        const link = document.createElement("a");
+        link.href = url;
+
+        // Create a date+time string in the local timezone to use as the filename
+        const date = new Date();
+        const localTime = new Date(
+          date.getTime() - date.getTimezoneOffset() * 60_000
+        )
+          .toISOString()
+          .replace(/\..+$/, "")
+          .replace("T", "_")
+          .replaceAll(":", "-");
+
+        link.download = `demo_${localTime}.mcap`;
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+        URL.revokeObjectURL(url);
+        setShowDownloadInfo(true);
+      })();
+    },
+    [state]
+  );
+
+  return (
+    <section className={styles.container}>
+      <div className={styles.column}>
+        <header>
+          <h2>Try it out</h2>
+          <div className={styles.subhead}>
+            Record an MCAP file right from your browser. Select sensors, create
```

Comment:
```suggestion
            Record xyz information in an MCAP file, and download it for a hands-on look at the MCAP file format.
```

And separate line: "Select sensor data to record in your MCAP file. All data is recorded and saved locally."